src/marqo/tensor_search/validation.py

```diff
@@ -394,15 +399,63 @@ def validate_dict(field: str, field_content: Dict, is_non_tensor_field: bool, ma
                     f"See `{marqo_docs.mappings()}` for more info on object fields. "
                 )
         else:
-            # Index is unstructured, so it must be in mappings.
-            raise InvalidArgError(
-                f"The field {field} must be in the add_documents `mappings` parameter to use it as an object field type. "
-                f"However, the `mappings` provided is: {mappings}. Please change the field content or add the field to `mappings`. "
-                f"See `{marqo_docs.mappings()}` for more info on object fields."
-            )
+            # Index is unstructured, check if it is a map numeric field
+            if marqo_index_version_lt_2_9_0:
+                raise InvalidArgError(
+                    f"Unsupported map field. "
+                    f"In Marqo versions prior to 2.9.0, only custom vector fields are supported as map fields "
+                    f"and must be declared in the `mappings` parameter. "
+                    f"Only indexes created with Marqo 2.9.0 or above support map numerical fields."
+                )
+            try:
+                field_content = validate_map_numeric_field(field_content)
+            except Exception:
+                raise InvalidArgError(
+                    f"The field '{field}' contains a dictionary value. "
```

Comment:
I think the errors coming from the validate function are fine to raise without more details

Reply:
done

Reply:
done